### PR TITLE
Decompile playback.c

### PR DESCRIPTION
**Pull request by JoshDuMan**

### CODE REVIEW

File: src/code/playback.c [hunk 1]
```diff
@@ -0,0 +1,786 @@
+#include <ultra64.h>
+#include <global.h>
+
+#ifdef NON_MATCHING
+// 630 score, primarily regalloc left. Issues seem based around the usage of pan.
+// Currently the array of pan and reverb solves a lot, but doesn't make a ton of logical sense.
+void note_set_vel_pan_reverb(Note* note, NoteSubEu* sub, Reverb* reverb) {
```
Comment (krimtonz) [inline src/code/playback.c:7]: This does not fit our naming scheme. 
The suggestion would be something that fits our naming scheme, 
```suggestion
void Playback_NoteVelPanReverbSet(Note* note, NoteSubEu* sub, Reverb* reverb) {
```

File: include/z64audio.h [hunk 2]
```diff
@@ -0,0 +1,495 @@
...
+    u32 drum4;
+    // u8 releaseRate;
+    // u8 pan;
+    // u8 loaded;
+    // AudioBankSound sound;
+    // AdsrEnvelope *envelope;
+} Drum; // OOT size = 0x8, TODO figure out what is still used.
+
+typedef struct {
+    u32 ootNew;
```
Comment (krimtonz) [inline include/z64audio.h:107]: Personally I'm not a fan of the naming scheme here, should keep up with our unknown naming scheme, comments can be made where differences apply from SM64 
```suggestion
    u32 unk_00; // new in OOT from sm64
    u32 unk_04; // new in OOT from sm64
```

File: src/code/playback.c [hunk 3]
```diff
@@ -0,0 +1,786 @@
...
+        !(ret = alloc_note_from_decaying(&seqLayer->seqChannel->seqPlayer->notePool, seqLayer)) &&
+        !(ret = alloc_note_from_decaying(&gNoteFreeLists, seqLayer)) &&
+        !(ret = alloc_note_from_active(&seqLayer->seqChannel->notePool, seqLayer)) &&
+        !(ret = alloc_note_from_active(&seqLayer->seqChannel->seqPlayer->notePool, seqLayer)) &&
+        !(ret = alloc_note_from_active(&gNoteFreeLists, seqLayer))) {
+        goto null_return;
+    }
+    return ret;
+
+null_return:
```
Comment (krimtonz) [inline src/code/playback.c:758]: Replacing the goto null_returns with
```
seqLayer->unusedEu0b8 = 1;
return NULL;
```

does not match?

File: src/code/playback.c [hunk 4]
```diff
@@ -0,0 +1,786 @@
...
+Note* pop_node_with_value_less_equal(AudioListItem* list, s32 limit) {
+    AudioListItem* cur = list->next;
+    AudioListItem* best;
+
+    if (cur == list) {
+        return NULL;
+    }
+
+    best = cur;
+    for (; cur != list; cur = cur->next) {
```
Comment (krimtonz) [inline src/code/playback.c:579]: Could the for loop be 
```suggestion
    for (cur = list->next, best = cur; cur != list; cur = cur->next) {
```
And remove 
```
if(cur == list) {
    return NULL;
}

best = cur;
```

File: src/code/playback.c (hunk 3 above)
Comment (JoshDuMan) [reply src/code/playback.c:758]: So I'll give this one a retry, but that is actually what SM64 has but it didn't match in this situation. For some reason it doesn't know to merge the tails here (perhaps there's a compiler flag for this?)

Comment (JoshDuMan) [reply src/code/playback.c:758]: Yeah, just checked and it doesn't merge tails for optimization. Maybe there's a way to that would trigger it naturally, but I have been unable to.

File: src/code/playback.c (hunk 4 above)
Comment (JoshDuMan) [reply src/code/playback.c:579]: I can't remove the conditional, but was able to put the rest in the for and remove best = cur;

**Comment by JoshDuMan** [comment]
Addressed all comments to this point.

**Pull request merged**